frontends/relay-futil/dahlia_functions.py

```diff
@@ -0,0 +1,125 @@
+import subprocess
+
+from futil_ast import *
+
+
+def lower_dahlia_program(prog, component_name):
+    '''
+    Takes in a string that represents a Dahlia program, lowers it to FuTIL, and applies the `externalize` pass.
+    This is just for experimental purposes, and needs to be replaced.
+    More bluntly, this does the following:
+    1. Copies dahlia program `prog` to a temporary file `temp.fuse`.
+       $ echo `program_string` > temp.fuse
+
+    2. Lowers `temp.fuse` to FuTIL with the name changed to `component_name`, and saves it in `lowered.futil`.
+       $ ./fuse temp.fuse --lower -b=futil -n=component_name > lowered.futil
+
+    3. Runs the 'externalize' pass on the `lowered.futil` file.
+       $ cargo run -- lowered.futil -p externalize > temp.futil
+
+    4. Copies the output from `lowered.futil`, except for the first line (we don't want another copy of the import).
+
+    TODO(cgyurgyik): As you'll see below, this only works on my local machine.
+                     I've explicitly removed errors with `2>/dev/null` so they aren't inserted
+                     to the file as well. However, this makes debugging difficult as well.
+    '''
+    program_string = "\""
+    for line in prog.splitlines():
+        program_string += f'{line}\n'
+    program_string += "\""
+    no_err = "2>/dev/null"
+    command = \
+        f"""
+        echo {program_string} > temp.fuse &&
+        /Users/cgyurgyik/Projects/dahlia/fuse temp.fuse --lower -b=futil -n={component_name} > lowered.futil {no_err} -l error &&
```

Comment:
the thing to do here is to expect that `dahlia` or `fuse` is available on the path. If you want, you can also allow the path to be passed in with an environment variable. If you want to support that, I would check to see if a particular environment variable is set, if it is use that otherwise default to either `dahlia` or `fuse`

Reply:
Ok I believe I've fulfilled this, and updated the README accordingly. PTAL.